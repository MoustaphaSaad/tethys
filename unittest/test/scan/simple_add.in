proc main
	i32.load r0 -1
	i32.load r1 2
	i32.add r0 r1
	halt
end