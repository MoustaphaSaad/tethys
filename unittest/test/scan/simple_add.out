line: 1, col: 0, kind: "PROC" str: "proc"
line: 1, col: 5, kind: "<ID>" str: "main"
line: 2, col: 2, kind: "<COMMENT>" str: "this is a comment"
line: 3, col: 2, kind: "i32.load" str: "i32.load"
line: 3, col: 11, kind: "R0" str: "r0"
line: 3, col: 14, kind: "<INTEGER>" str: "-1"
line: 4, col: 2, kind: "i32.load" str: "i32.load"
line: 4, col: 11, kind: "R1" str: "r1"
line: 4, col: 14, kind: "<INTEGER>" str: "2"
line: 5, col: 2, kind: "i32.add" str: "i32.add"
line: 5, col: 10, kind: "R0" str: "r0"
line: 5, col: 13, kind: "R1" str: "r1"
line: 6, col: 2, kind: "HALT" str: "halt"
line: 7, col: 1, kind: "END" str: "end"
